numpy/lib/tests/test_polynomial.py

```diff
@@ -1,93 +1,87 @@
-from __future__ import division, absolute_import, print_function
-
-'''
->>> p = np.poly1d([1.,2,3])
->>> p
-poly1d([ 1.,  2.,  3.])
->>> print(p)
-   2
-1 x + 2 x + 3
->>> q = np.poly1d([3.,2,1])
->>> q
-poly1d([ 3.,  2.,  1.])
->>> print(q)
-   2
-3 x + 2 x + 1
->>> print(np.poly1d([1.89999+2j, -3j, -5.12345678, 2+1j]))
-            3      2
-(1.9 + 2j) x - 3j x - 5.123 x + (2 + 1j)
->>> print(np.poly1d([-3, -2, -1]))
-    2
--3 x - 2 x - 1
-
->>> p(0)
-3.0
->>> p(5)
-38.0
->>> q(0)
-1.0
->>> q(5)
-86.0
-
->>> p * q
-poly1d([  3.,   8.,  14.,   8.,   3.])
->>> p / q
-(poly1d([ 0.33333333]), poly1d([ 1.33333333,  2.66666667]))
->>> p + q
-poly1d([ 4.,  4.,  4.])
->>> p - q
-poly1d([-2.,  0.,  2.])
->>> p ** 4
-poly1d([   1.,    8.,   36.,  104.,  214.,  312.,  324.,  216.,   81.])
-
->>> p(q)
-poly1d([  9.,  12.,  16.,   8.,   6.])
->>> q(p)
-poly1d([  3.,  12.,  32.,  40.,  34.])
-
->>> np.asarray(p)
-array([ 1.,  2.,  3.])
->>> len(p)
-2
-
->>> p[0], p[1], p[2], p[3]
-(3.0, 2.0, 1.0, 0)
-
->>> p.integ()
-poly1d([ 0.33333333,  1.        ,  3.        ,  0.        ])
->>> p.integ(1)
-poly1d([ 0.33333333,  1.        ,  3.        ,  0.        ])
->>> p.integ(5)
-poly1d([ 0.00039683,  0.00277778,  0.025     ,  0.        ,  0.        ,
-        0.        ,  0.        ,  0.        ])
->>> p.deriv()
-poly1d([ 2.,  2.])
->>> p.deriv(2)
-poly1d([ 2.])
-
->>> q = np.poly1d([1.,2,3], variable='y')
->>> print(q)
-   2
-1 y + 2 y + 3
->>> q = np.poly1d([1.,2,3], variable='lambda')
->>> print(q)
-        2
-1 lambda + 2 lambda + 3
-
->>> np.polydiv(np.poly1d([1,0,-1]), np.poly1d([1,1]))
-(poly1d([ 1., -1.]), poly1d([ 0.]))
-
-'''
-import numpy as np
-from numpy.testing import (
-    run_module_suite, assert_, assert_equal, assert_array_equal,
-    assert_almost_equal, assert_array_almost_equal, assert_raises, rundocs
-    )
-
+from __future__ import absolute_import, division, print_function
 
-class TestDocs(object):
-    def test_doctests(self):
-        return rundocs()
+import numpy as np
+from numpy.testing import (assert_, assert_almost_equal,
+                           assert_array_almost_equal, assert_array_equal,
+                           assert_equal, assert_raises, run_module_suite)
+
+
+def assert_str_equal(actual, desired, err_msg='', verbose=True):
+    assert_equal(str(actual), desired, err_msg, verbose)
+
+
+def assert_repr_equal(actual, desired, err_msg='', verbose=True):
+    assert_equal(repr(actual), desired, err_msg, verbose)
+
+
+class TestPolynomial(object):
+    def test_poly1d_str_and_repr(self):
+        # GH9416: Convert doctests to regular tests.
+        p = np.poly1d([1., 2, 3])
+        assert_repr_equal(p, 'poly1d([1., 2., 3.])')
```

Comment:
This is only one character shorter than `assert_equal(repr(p), ...)`, and the latter is clearer in my opinion.